packages/flutter/test/material/tabs_test.dart

```diff
@@ -4141,6 +4141,38 @@ void main() {
     expect(tester.getCenter(find.byKey(lastTabKey)).dx, equals(750.0));
   });
 
+  testWidgets('DefaultTabController changes does not recreate PageController', (WidgetTester tester) async {
+    // This is a regression test for https://github.com/flutter/flutter/issues/134253.
+    Widget buildFrame(int length) {
+      return boilerplate(
+        child: DefaultTabController(
+          length: length,
+          initialIndex: length - 1,
+          child: TabBarView(
+            physics: const TestScrollPhysics(),
+            children: List<Widget>.generate(
+              length,
+              (int index) {
+                return Center(child: Text('Page $index'));
+              },
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildFrame(10));
+    PageView pageView = tester.widget(find.byType(PageView));
+    final PageController pageController1 = pageView.controller;
+
+    // Rebuild with a new default tab controller with more tabs.
+    await tester.pumpWidget(buildFrame(15));
+    pageView = tester.widget(find.byType(PageView));
+    final PageController pageController2 = pageView.controller;
+
+    expect(pageController1, equals(pageController2));
```

Comment:
also check whether the index is correct?

Reply:
@chunhtai I consider it but I found it confusing to add this check in the test because, with or without the PR, the index will not change (`DefaultTabCotnroller.initialIndex` is read once when initializing the `PageView`). 
I don't know if this is the correct behavior?

Reply:
> I thought the jumpTo will make it jump to whatever less rebuild index which is 14

It won't because changing the `DefaultTabController.initialPage` has no impact on the page view current page index. Technically, it is because `_DefaultTabControllerState.didUpdateWidget` does not check if `initialPage` was changed. I would say it makes sense because changing the `initialIndex` does not mean changing the current index, but I don't have a strong opinion there.

For the moment, I have not find a case were the `jumpTo` added in this PR would be necessary (but it would if  we decided that `_DefaultTabControllerState.didUpdateWidget` should update the selected page when `DefaultTabController.initialPage` is changed.

Reply:
ah yes. in this case, can you shorten the length to force it to update the index in controller?

for example, if current index is 10, you rebuild DefaultTabController with length 5, it should force the index to be 4

Reply:
That's exactly what I tried this morning :smile: .
In this case the index changes so I thought it will be ok to update the test accordingly... until I checked what happened if I remove the call to `jumpTo`: the index still change in this case because it is updated earlier (because of the logic in `_DefaultTabControllerState.didUpdateWidget` and because the viewport dimension changed lead to a call to `_PagePosition.forcePixels` which correctly updated the scroll position pixels on which the page computation is based).

Reply:
You meant viewportFraction change? I can't figure out how the method is called in the update.

Either way, I think I will still put the check in this test. Even though it pass without the jumpTo, as long as it is checking correct behavior, it is worth adding there to prevent future regressions.

Reply:
I totaly agree that it is worth adding this to prevent future regressions. I have updated the PR accordingly.
Many thanks for sharing your thoughts.

> You meant viewportFraction change?

No, I meant when the length of the `TabController` is shorten, it has an impact on the page view content dimensions and the logic in `_PagePosition.applyContentDimensions` will take care of updating the page controller position (I was wrong mentionning `forcePixels` it is `correctPixels` which is called).

For posterity :smile: , the way to get deeper on this is to run the test in this PR. Then comment the call to `jumpTo` added in this PR and see that the test still pass. To understand why put a break point in `ScrollPosition.correctPixels` and re-run the test. The call stack will be the following one:

![Capture d’écran du 2023-10-04 20-04-04](https://github.com/flutter/flutter/assets/840911/75cf47ef-182d-448e-a0ed-95ae7b69711f)

Reply:
ah this totally makes sense now, thanks for explanation.